contributors/guide/pull-requests.md

```diff
@@ -590,6 +590,26 @@ at once to that file.
 * Can the file be improved further?
 * Does the trivial edit greatly improve the quality of the content?
 
+## Fixing linter issues
+
+Kubernetes has a set of linter checks. Some of those must pass in the entire
+code base, some must pass in new or modified code, and some are merely hints
+to developers how to improve their code.
+
+Please do not create Pull Requests for issues found by linters without first
+reaching out to maintainers to determine whether there is sufficient interest
+in fixing such issues. It does not matter whether the linter is enabled in
+Kubernetes or not:
+- If a linter *is* enabled in
+  [golangci.yaml](https://github.com/kubernetes/kubernetes/blob/master/hack/golangci.yaml),
+  then it has already been determined that sweeping changes in the existing
+  code aren't necessary.
+- If a linter *is not* enabled, then it might not be important enough.
```

Comment:
If people start asking about the same linter checks over and over again, then we might want to add a section to `golangci.yaml` where we document "rejected" linters and the rationale. But this might not become a real issue - we can worry about this when it starts to happen.

Reply:
it seems to happen pretty regularly

Reply:
maybe guidelines around which linters to propose would be helpful ... my opinion is that we should generally only consider linters which actually improve code correctness (not style, format, constant references, etc)

Reply:
I'm guilty of approving those PRs when they improve readability (which is very subjective) or error messages (e.g. all the testify stuff).

Reply:
I think the testify linter improves code quality. Tests are also code, and an assertion that is more readable because it matches the intent of the test and then produces a more informative failure message is better than `require.True(<some boolean expression>)`.

Reply:
Of course, such a `require.True` wasn't *wrong* and thus strictly speaking didn't need to be fixed. But if at some point a test fails because of a regression, we want the test to be readable by whoever ends up debugging it and we want the useful failure message in our triage tool and the issue. "Expected true, got false" is not helpful at that point.

There's another aspect: a lot of new tests are written by looking at existing tests, perhaps even via cut-and-paste. It's easier for new contributors when they find high-quality tests to learn from, instead of later being told via hints that they could have done something better.

Reply:
> maybe guidelines around which linters to propose would be helpful

Added.